[protobuf-cpp] Initial plan and tests

MindNumbing opened the pull request:
Signed-off-by: Steven Marshall <SMarshall@chef.io>

Commit:
[protobuf-cpp] Initial plan and tests

Signed-off-by: Steven Marshall <SMarshall@chef.io>

gavindidrichsen commented on controls/protobuf-cpp_works.rb:
```diff
@@ -0,0 +1,36 @@
+  hab_pkg_path = command("hab pkg path #{plan_ident}")
+  describe hab_pkg_path do
+    its('exit_status') { should eq 0 }
+    its('stdout') { should_not be_empty }
```
Extra sterr check, its('stderr') { should be_empty }

gavindidrichsen commented on controls/protobuf-cpp_works.rb:
```diff
@@ -0,0 +1,36 @@
+    its('exit_status') { should eq 0 }
+  end
+
+  protoc_works = command("/bin/protoc --version")
```
Could you use the full /hab/pkgs/core/.../protoc path here instead to ensure that the actual installation is getting tested.  You could reuse the File.join above, I think.

gavindidrichsen reviewed (commented):
Just a couple minor updates, but otherwise looks good

Commit:
[protobuf-cpp] Various refactors of existing code

Added inspec.lock to .gitignore
Updated README.md to latest binary format
Added plan_name to attributes.yml
Updates controls to add more prescriptive version checking.
Updated inspec.yml to latest version

Signed-off-by: Steven Marshall <SMarshall@chef.io>

skpaterson reviewed (approved):
Thanks @MindNumbing - looks like all comments are addressed.
